client/dev-tools-promo/index.tsx

```diff
@@ -1,15 +1,30 @@
+import { FEATURE_SFTP, WPCOM_FEATURES_ATOMIC } from '@automattic/calypso-products';
 import page, { Context as PageJSContext } from '@automattic/calypso-router';
 import { makeLayout, render as clientRender } from 'calypso/controller';
 import { navigation, siteSelection, sites } from 'calypso/my-sites/controller';
 import { siteDashboard } from 'calypso/sites-dashboard-v2/controller';
 import { DOTCOM_DEVELOPER_TOOLS_PROMO } from 'calypso/sites-dashboard-v2/site-preview-pane/constants';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import { devToolsPromo } from './controller';
 
 const redirectForNonSimpleSite = ( context: PageJSContext, next: () => void ) => {
 	const state = context.store.getState();
 	const site = getSelectedSite( state );
-	if ( site && site.jetpack && ! site.plan?.expired ) {
+	const siteId = site?.ID ?? null;
+	const isJetpackSite = !! site?.jetpack;
+	const hasAtomicFeature = siteHasFeature( state, siteId, WPCOM_FEATURES_ATOMIC );
+	const hasSftpFeature = siteHasFeature( state, siteId, FEATURE_SFTP );
```

Comment:
Curious why we need to explicitly check for this? I believe SFTP is only possible if the site is Atomic, so shouldn't `hasAtomicFeature` be already sufficient?

Reply:
🤔 Actually, I think maybe we can just check for SFTP feature access. Example existing pattern [here](https://github.com/Automattic/wp-calypso/blob/df0ba7adcafb6a563c147e7689059192f0e24fa4/client/my-sites/hosting/main.js#L295).

In theory, some plans can go Atomic (at one point all plans were going to be able to go Atomic): fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qsrngherf%2Spynff%2Qjcpbz%2Qsrngherf.cuc%3Se%3Q3919886s%23508%2Q513-og

but not all sites with the Atomic feature have SFTP feature access:
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qsrngherf%2Spynff%2Qjcpbz%2Qsrngherf.cuc%3Se%3Q3919886s%23995%2Q1000-og

So, SFTP access is probably the better signal for access to Hosting Configuration tools.